src/datasources/db/v2/database-migrator.service.spec.ts

```diff
@@ -0,0 +1,140 @@
+    // by GitHub actions
+    const isCIContext = process.env.CI?.toLowerCase() === 'true';
+    const baseConfiguration = configuration();
+    const testConfiguration: typeof configuration = () => ({
+      ...baseConfiguration,
+      db: {
+        ...baseConfiguration.db,
+        migrator: {
+          numberOfRetries: NUMBER_OF_RETRIES,
+          migrationsExecute: false,
+          retryAfter: 10,
+        },
+        connection: {
+          postgres: {
+            ...baseConfiguration.db.connection.postgres,
+            ssl: {
+              ...baseConfiguration.db.connection.postgres.ssl,
+              enabled: !isCIContext,
+              caPath: join(__dirname, '../../../../db_config/test/server.crt'),
+            },
+          },
+        },
+      },
+    });
+
+    moduleRef = await Test.createTestingModule({
+      imports: [
+        TestLoggingModule,
+        ConfigurationModule.register(testConfiguration),
+        ConfigModule,
+        TestPostgresDatabaseModuleV2,
+      ],
+      providers: [DatabaseMigrator],
+    }).compile();
+
+    databaseMigratorService = moduleRef.get<DatabaseMigrator>(DatabaseMigrator);
+    postgresDatabaseService = moduleRef.get<PostgresDatabaseService>(
+      PostgresDatabaseService,
+    );
+    loggingService = moduleRef.get<ILoggingService>(LoggingService);
+
+    connection = postgresDatabaseService.getDataSource();
+  });
+
+  afterAll(async () => {
+    await moduleRef.close();
+  });
+
+  beforeEach(() => {});
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('migrate()', () => {
+    it('Should log the start and end of the migration process', async () => {
+      jest.spyOn(loggingService, 'info');
+
+      await databaseMigratorService.migrate();
+
+      expect(loggingService.info).toHaveBeenCalledWith(
+        'Migrations: Running...',
+      );
+      expect(loggingService.info).toHaveBeenCalledWith('Migrations: Finished.');
+      expect(loggingService.info).toHaveBeenCalledTimes(3);
+    });
+
+    it('Should run migration if no lock exists', async () => {
+      jest.spyOn(connection, 'query').mockImplementation(jest.fn());
+
+      await databaseMigratorService.migrate();
+
+      expect(connection.runMigrations).toHaveBeenCalled();
+      expect(connection.query).toHaveBeenCalledWith(insertLockQuery, [1]);
+
+      expect(connection.query).toHaveBeenCalledWith(truncateLockQuery);
+    });
+
+    it('Should throw an error if retries are exhausted', async () => {
+      connection.query = jest.fn().mockResolvedValue([{ id: 1, status: 1 }]);
+
+      await expect(databaseMigratorService.migrate()).rejects.toThrow(
+        'Migrations: Migrations are still running in another instance!',
+      );
+
+      expect(loggingService.info).toHaveBeenCalledWith(
+        'Migrations: Running in another instance...',
+      );
+      expect(loggingService.info).toHaveBeenCalledTimes(2);
```

Comment:
You're only checking the call value once here, but expecting it to be called twice. We should expect both and use `toHaveBeenNthCalledWith`.

The same can be done in a test below - line 134.

Reply:
In my opinion, a good test should have at most three assertions, but I made this change for the sake of consistency. We should discuss this further later.
https://github.com/safe-global/safe-client-gateway/pull/1988/commits/2ea0e5d8a752fd31cff745ff3490d8fe8134ddc7

Reply:
If your are expecting n calls, I think it's beneficial to also expect _what_ those calls are. For example, the arguments may change later but the number of calls may not. There are other expectations in this suite that are the same as my intial comment. However, if you disagree, you can leave it as is.